tests/helpers/test_script.py

```diff
@@ -1053,6 +1053,7 @@ async def test_multiple_runs_wait(hass: HomeAssistant, action_type) -> None:
         hass.states.async_set("switch.test", "on")
         hass.async_create_task(script_obj.async_run(context=Context()))
         await asyncio.wait_for(wait_started_flag.wait(), 1)
+        await asyncio.sleep(0)
```

Comment:
There are other `asyncio.wait_for` in tests, why do we need to add a sleep in this test?

Reply:
The `wait_for` behavior has changed in python 3.12 The underlying implementation no longer creates a task so there was an implicit delay happening before

We should replace all usage of `asyncio.wait_for` with `asyncio.timeout` since its behavior is much more predictable

In fact `asyncio.wait_for` in python 3.12 is effectively a wrapper around `asyncio.timeout` now

Reply:
> The wait_for behavior has changed in python 3.12. The underlying implementation no longer creates a task so there was an implicit delay happening before

Right, so there's some risk that other tests calling `asyncio.wait_for` are now flaky?